scalyr_agent/monitor_utils/k8s.py

```diff
@@ -322,9 +337,11 @@ def shallow_copy(self):
 
         return result
 
-    def purge_expired( self, access_time ):
-        """
-            removes any items from the store who haven't been accessed since `access_time`
+    def purge_unused(self, access_time, soft_purge=False):
```

Comment:
I think this should probably be split in to two separate functions.

`purge` was used as name to explicitly refer to removing items from the cache that were no longer in use - as a form of garbage collection essentially.  This is no longer as clear because `mark expired` functionality has now been mixed in also.

With the way this is called by `update_cache` for soft purges, it's always going to mark ever item as stale (because it passes in the current time as the `access_time` so `obj.access_time < access_time` will be true for everything).

That being the case, it'd be clearer to have a `mark_as_expired` function that goes over every item in objects and marked it as expired.  That's what is happening here anyway for soft purges, it's just not clear. 

`update_cache` always sleeps for `cache_expire_secs` so that's an appropriate place for `mark_as_expired` to be called.

Then this function can always just remain a hard purge, for when `cache_purge_secs` expires, and `update_cache` is for what happens when `cache_expire_secs` expires.

Reply:
Good suggestion that helps long term maintenance. 👍